src/router.ts

```diff
@@ -366,7 +427,8 @@ export class Router {
       }
 
       this.navigation.push(navModel);
-      this.navigation = this.navigation.sort((a, b) => a.order - b.order);
+      // this is a potential error / inconsistency between browsers
```

Comment:
I was unclear here. Only returning `0` will cause inconsistency behavior, per spec. From MDN:

> If compareFunction(a, b) returns 0, leave a and b unchanged with respect to each other, but sorted with respect to all different elements. Note: the ECMAscript standard does not guarantee this behaviour, and thus not all browsers (e.g. Mozilla versions dating back to at least 2003) respect this.

Because of that, if we have a list of route configs that all have `nav: true`, it will fall in this `return 0` behavior, thus may yield inconsistency